packages/file_selector/file_selector_macos/macos/Classes/FileSelectorPlugin.swift

```diff
@@ -73,7 +73,7 @@ public class FileSelectorPlugin: NSObject, FlutterPlugin {
       configure(panel: panel, with: arguments)
       configure(openPanel: panel, with: arguments, choosingDirectory: choosingDirectory)
       panelController.display(panel, for: viewProvider.view?.window) { (selection: [URL]?) in
-        if (choosingDirectory) {
+        if (choosingDirectory && panel.allowsMultipleSelection == false) {
```

Comment:
You've correctly changed the Dart side to expect a list, but are not sending one, so this will crash.

Reply:
Change applied, thanks.